pkg/client/keystone/keystone.go

```diff
@@ -2,20 +2,94 @@ package keystone
 
 import (
 	"bytes"
+	"crypto/tls"
+	"crypto/x509"
 	"encoding/json"
 	"fmt"
+	"io"
 	"io/ioutil"
 	"net/http"
 
+	contrail "github.com/Juniper/contrail-operator/pkg/apis/contrail/v1alpha1"
+	"github.com/Juniper/contrail-operator/pkg/certificates"
 	"github.com/Juniper/contrail-operator/pkg/client/kubeproxy"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-func NewClient(client *kubeproxy.Client) *Client {
-	return &Client{client: client}
+// NewClient prepares keystone client based on properties of passed keystone CRD.
+func NewClient(kubClient client.Client, scheme *runtime.Scheme, config *rest.Config, k *contrail.Keystone) (*Client, error) {
+	if k.Spec.ServiceConfiguration.ExternalAddress != "" {
+		caCertificate := certificates.NewCACertificate(kubClient, scheme, k, k.GetName())
+		caBundle, _ := caCertificate.GetCaCert()
```

Comment:
do we want to ignore this error? i think that in case of error we should perform another action like retry since you already created a CACert and it cannot be retrieved or maybe error should be returned

Reply:
My understanding was if the caBundle will be nill (in case of  GetCaCert() error) the newExtKeystoneClient will not add anything to rootCA. If the caBundle is required for endpoint verification the client will return error. If endpoint cert is signed by authority defined in operating system ca bundle then request still should succeed even getting internal ca failed.